f8a_worker/workers/downstream.py

```diff
@@ -1,253 +0,0 @@
-import os
-
-from f8a_worker.base import BaseTask
-from f8a_worker.enums import EcosystemBackend
-from f8a_worker.errors import TaskError
-from f8a_worker.models import Ecosystem
-from f8a_worker.pulp import Pulp
-from f8a_worker.schemas import SchemaRef
-from f8a_worker.utils import TimedCommand, MavenCoordinates
-from f8a_worker.workers.anitya import RH_MVN_DISTRO_NAME, RH_MVN_GA_REPO, RH_RPM_DISTRO_NAME
-
-
-# Test hook: the worker tests mock this out,
-# while integration tests provide a full Anitya instance
-def _query_anitya_url(host_url, api_path):
-    return requests.get(host_url + api_path)
-
-
-class DownstreamUsageTask(BaseTask):
-    """Queries Red Hat's internal toolchain for downstream component usage
-
-    - queries Anitya for downstream package names
-    - uses the package name and component version to query:
-      - Brew for internal SRPM and build details
-      - the Pulp CDN for redistribution details
-    """
-    _analysis_name = 'redhat_downstream'
-    schema_ref = SchemaRef(_analysis_name, '2-2-1')
-
-    _backend_to_anitya_ecosystem = {
-        EcosystemBackend.npm: 'npm',
-        EcosystemBackend.maven: 'maven',
-        EcosystemBackend.pypi: 'pypi',
-        EcosystemBackend.rubygems: 'rubygems',
-        EcosystemBackend.nuget: 'nuget'
-    }
-
-    _ecosystem_to_prefix = {
-        'npm': 'nodejs',
-        'pypi': 'python',
-        'rubygems': 'rubygem'
-    }
-
-    # Give CLI 10 minutes to retrieve results
-    _BREWUTILS_CLI_TIMEOUT = 600
-
-    def _get_artifact_hash(self, algorithm=None):
-        wr = self.parent_task_result('digests')
-        if wr:
-            for details in wr['details']:
-                if details.get('artifact'):
-                    return details[algorithm or 'md5']
-        return None
-
-    @staticmethod
-    def _prefix_package_name(name, ecosystem):
-        prefix = DownstreamUsageTask._ecosystem_to_prefix.get(ecosystem, '')
-        if prefix:
-            return '{p}-{n}'.format(p=prefix, n=name)
-
-        return name
-
-    def _fetch_anitya_project(self, ecosystem, package):
-        eco_model = self.storage.get_ecosystem(ecosystem)
-        backend = self._backend_to_anitya_ecosystem.get(eco_model.backend, None)
-        if backend is None:
-            raise ValueError('Don\'t know how to add ecosystem {e} with backend {b} to Anitya'.
-                             format(e=ecosystem, b=eco_model.backend))
-        api_path = '/api/by_ecosystem/{e}/{p}/'.format(e=ecosystem, p=package)
-        anitya_url = self.configuration.ANITYA_URL
-        try:
-            return _query_anitya_url(anitya_url, api_path)
-        except (requests.HTTPError, requests.ConnectionError):
-            msg = 'Failed to contact Anitya server at {}'
-            self.log.exception(msg.format(self.configuration.ANITYA_URL))
-        return None
-
-    def _get_cdn_metadata(self, srpm_filename):
-        """Try to retrieve Pulp CDN metadata"""
-        try:
-            pulp = Pulp()
```

Comment:
You can remove also all the Pulp related code like [pulp.py](https://github.com/fabric8-analytics/fabric8-analytics-worker/blob/master/f8a_worker/pulp.py) since it's been used by this task only.